src/ucp/api/ucp.h

```diff
@@ -1188,6 +1195,17 @@ typedef struct ucp_worker_params {
      */
     uint64_t                flags;
 
+    /**
+     * Tracing and analysis tools can identify the worker using this name.
+     * This value is optional.
+     * The worker name may differ from the name specified in the parameters to
+     * avoid potential name conflicts. Please use @ref ucp_worker_query to get
+     * the actual name of the worker.
+     * If @ref UCP_WORKER_PARAM_FIELD_NAME is not set in the field_mask, the
+     * worker will have a default name.
```

Comment:
rework of lines 1198-1205:

Tracing and analysis tools can identify the worker using this name.  To retrieve the worker's name, use @ref ucp_worker_query, as the name you supply may be changed by UCX under some circumstances, e.g. a name conflict.  This field is only assigned if you set @ref UCP_WORKER_PARAM_FIELD_NAME in the field mask.  If not, then a default unique name will be created for you.

Reply:
The comment in ucp_params_t for the name field needs to be revisited too

Reply:
Updated description here and in ucp_params_t.

Reply:
We agreed to not describe the possible name changes in different cases in this discussion https://github.com/openucx/ucx/pull/6368#discussion_r582145059
... to avoid creating dependency on implementation. User can get the actual name using query method which is mentioned in the description.